### PR TITLE
doc: add doc for FileMap

### PR DESCRIPTION
**Pull request by joneugster**

### CODE REVIEW

**Comment by joneugster** [comment]
I'm not sure if Pr's only adding documentation, like this one, are generally well received. If you don't have the time/interest to review such Pr's please just close it!

File: src/Lean/Data/Position.lean [hunk 1]
```diff
@@ -29,9 +29,16 @@ instance : ToExpr Position where
...
+/-- Content of a file together with precalculated positions of newlines. -/
 structure FileMap where
+  /-- The content of the file. -/
   source    : String
+  /-- The positions of newline characters.
+  The first entry is always `0` and the last always the index of the last character.
+  In particular, if the last character is a newline, that index will appear twice. -/
   positions : Array String.Pos
+  /-- The line indices. Has the same length as `positions` and is always of the form
+  `[1, 2, …, n-1, n-1]`. -/
```
Comment (david-christiansen) [inline src/Lean/Data/Position.lean:41]: I'm not sure I understand this specification here. There's a few things about it that I find confusing:

1. (minor) It uses list syntax for an array
2. I'm not sure what a line index is. Can you define the term? Or what about calling it "line number associated with each mapped newline"?

**Comment by david-christiansen** [comment]
Thanks for the PR!

File: src/Lean/Data/Position.lean (hunk 1 above)
Comment (joneugster) [reply src/Lean/Data/Position.lean]: updated!

One point is that `lines` seems to be pretty redundant, as far as I can tell.

Comment (Kha) [reply src/Lean/Data/Position.lean]: Yes, if you want to do a follow-up PR removing it, that would be a strict improvement I'd say!

**Review by david-christiansen — APPROVED** [review]

**Pull request merged**